### PR TITLE
spacetime local clear subcommand

### PR DESCRIPTION
**Pull request by jdetter**
# Description of Changes

Please describe your change, mention any related tickets, and so on here.

- I separated out the functionality from https://github.com/clockworklabs/SpacetimeDB/pull/699 where you can clear the local database. This PR does *not* break the ABI or API.
- Added a `spacetime local clear [--force]` which allows you to clear your local spacetimedb database without having to manually `rm -rf ~/.spacetime`

# API and ABI breaking changes

If this is an API or ABI breaking change, please apply the
corresponding GitHub label.

- This PR does not break the ABI or API

# Expected complexity level and risk

*How complicated do you think these changes are? Grade on a scale from 1 to 5,
where 1 is a trivial change, and 5 is a deep-reaching and complex change.*

*This complexity rating applies not only to the complexity apparent in the diff,
but also to its interactions with existing and future code.*

*If you answered more than a 2, explain what is complex about the PR,
and what other components it interacts with in potentially concerning ways.*

1

### CODE REVIEW

**Review by gefjon — APPROVED** [review]
Excellent!

File: crates/cli/src/subcommands/local.rs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            println!("Control node database: {}", control_node_dir.to_str().unwrap());
+        } else {
+            println!("Control node database: <not found>");
+        }
+        if worker_node_dir.exists() {
+            println!("Worker node database: {}", worker_node_dir.to_str().unwrap());
+        } else {
+            println!("Worker node database: <not found>");
+        }
+    }
```
Comment (bfops) [inline crates/cli/src/subcommands/local.rs:102]: this seems to be the `else` case for `control_node_dir.exists() || worker_node_dir.exists()` above? If so, wouldn't that mean that neither of the `exists` checks should pass..?

(also, is this overall `if` layer even needed? if neither directory exists, nothing really breaks, it just doesn't do much.)

File: crates/cli/src/subcommands/local.rs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        }
+
+        if control_node_dir.exists() {
+            std::fs::remove_dir_all(&control_node_dir)?;
+            println!("Deleted control node database: {}", control_node_dir.to_str().unwrap());
+        }
+        if worker_node_dir.exists() {
+            std::fs::remove_dir_all(&worker_node_dir)?;
+            println!("Deleted worker node database: {}", worker_node_dir.to_str().unwrap());
+        }
```
Comment (bfops) [inline crates/cli/src/subcommands/local.rs:90]: Similar comment to the one I made on https://github.com/clockworklabs/SpacetimeDB/pull/699#discussion_r1442135585. I would personally move to consolidate the "go" logic.

**Review by bfops — CHANGES_REQUESTED** [review]

File: crates/cli/src/subcommands/local.rs (hunk 2 above)
Comment (jdetter) [reply crates/cli/src/subcommands/local.rs:90]: I looked at the suggestion on the other PR and I think I've done what you're suggesting - whether you supply `--force` or not this is the codepath which removes the files. Could you give me an example of what you mean here?

File: crates/cli/src/subcommands/local.rs (hunk 1 above)
Comment (jdetter) [reply crates/cli/src/subcommands/local.rs:102]: Good catch, fixed 👍

File: crates/cli/src/subcommands/local.rs (hunk 2 above)
Comment (bfops) [reply crates/cli/src/subcommands/local.rs:90]: Ah, sorry, maybe just a bit left over - we still call `remove_dir_all` further up. I would replace the code on lines 68-78 with:
```rust
            let yes = input.trim().to_lowercase() == "y" || input.trim().to_lowercase() == "yes"
            if ! yes {
                println!("Aborting");
                return Ok(());
            }
```

**Review by bfops — APPROVED** [review]
LGTM!

**Pull request merged**